This doesn't work with meson yet. Instead copy the .dll to /bin for now..
--- gedit-3.32.2/gedit/gedit.c.orig	2019-05-15 02:59:28.000000000 +0200
+++ gedit-3.32.2/gedit/gedit.c	2019-06-05 09:25:57.235283800 +0200
@@ -40,7 +40,7 @@
 #include "gedit-dirs.h"
 #include "gedit-debug.h"
 
-#ifdef G_OS_WIN32
+#if 0
 #include <gmodule.h>
 static GModule *libgedit_dll = NULL;
 
@@ -113,12 +113,13 @@
 #ifdef OS_OSX
 	type = GEDIT_TYPE_APP_OSX;
 #else
-#ifdef G_OS_WIN32
+#if 0
 	if (!gedit_w32_load_private_dll ())
 	{
 		return 1;
 	}
-
+#endif
+#ifdef G_OS_WIN32
 	type = GEDIT_TYPE_APP_WIN32;
 #else
 	type = GEDIT_TYPE_APP_X11;
@@ -163,7 +164,7 @@
 		                     G_OBJECT (app)->ref_count);
 	}
 
-#ifdef G_OS_WIN32
+#if 0
 	gedit_w32_unload_private_dll ();
 #endif
 
